src/OpenTelemetry.ResourceDetectors.Azure/AzureVmMetaDataRequestor.cs

```diff
@@ -0,0 +1,40 @@
+// <copyright file="AzureVmMetaDataRequestor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Net.Http;
+using System.Text.Json;
+
+namespace OpenTelemetry.ResourceDetectors.Azure;
+
+internal sealed class AzureVmMetaDataRequestor : IAzureVmMetaDataRequestor
+{
+    private const string AzureVmMetadataEndpointURL = "http://169.254.169.254/metadata/instance/compute?api-version=2021-12-13&format=json";
```

Comment:
hard-coding api version here, looks incorrect to me. Should we have an option to pass API version in the constructor as optional parameter. It' expected that these APIs are backward compactible and should not break our implementation.

Reply:
Leaving an option to specify API version means we have to make sure our deserialization works for all of them (depending on the data format that is returned by specific API version). I am not sure if we should allow that. In future we could just update this version to use the latest.

Reply:
https://learn.microsoft.com/en-us/azure/virtual-machines/instance-metadata-service?tabs=windows#versioning